siege_utilities/geo/s2_utils.py

```
@@ -0,0 +1,698 @@
+optionally post-filters cells against the actual geometry using shapely.
+For a heavily concave polygon (e.g. Idaho), the bbox cover is loose; the
+post-filter step (default on) trims cells whose centers fall outside.
+
+Requires: s2sphere>=0.2.5.
+"""
+
+from __future__ import annotations
+
+import logging
+import math
+from typing import Iterable, List, Optional, Tuple
```

Ruff (0.15.12)

[warning] Import from `collections.abc` instead: `Iterable`

Import from `collections.abc`

(UP035)